### ISSUE TITLE
Bug: `perfectionist/sort-switch-case` breaks code with consecutive case statements preceeded by default

### ISSUE THREAD

**Issue by hugoAlt** [azat-io/eslint-plugin-perfectionist#191]
### Describe the bug

Thank you very much for the latest fixes regarding this rule.

However, in specific cases, where a `default` clause precedes consecutive `case` statements, `sort-switch-case` incorrectly reformats into code with different behavior.

### Code example

```typescript
let restMethod;
switch (restMethod) {
  case "POST":
  default:
  case "GET":
    console.log("Print me")
    break;
}
```

gets reformated into

```typescript
let restMethod;
switch (restMethod) {
  case "GET":
    console.log("Print me")
    break;
  case "POST":
  default:
}
```

The expected code should be:

```typescript
let restMethod;
switch (restMethod) {
  case "GET":
  case "POST":
  default:
    console.log("Print me")
    break;
}
``` 

### ESLint version

8.57.0

### ESLint Plugin Perfectionist version

3.1.0

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Thank you for your issue. I will fix this issue as soon as possible.

**Comment by azat-io** [comment]
Fixed in 4ee655eab15161777cbe3821b6513bcde2601859. Released in [v3.1.2](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.1.2)
